### DISCUSSION TITLE
In VS Code, a new Untitled-1 file is opened everytime copilot fetches new information

### DISCUSSION THREAD

**Discussion by efreibe** [community/community#45269] [Copilot Conversations]
### Select Topic Area

Bug

### Body

How to reproduce the issue:

1. Load any vscode project
2. Edit any file that has copilot enabled
3. Create a new Untitled-1 file and exit the file without saving
4. Go to the original file and make any change, a Untitled-1 file is created
5. Close Untitled-1 and repeat 4

Disabling copilot does not trigger that behavior.

https://user-images.githubusercontent.com/1061126/214412520-8fee8958-f6ea-45fa-b742-be644f7b5433.mov

https://user-images.githubusercontent.com/1061126/214412563-ba371b4a-7a0e-42e1-96ad-dbe3843bff01.mov

**Comment by CagriAldemir** [comment]
Hi @efreibe Did you solve this bug?

**Reply by efreibe** [reply]
Yes, this bug has been solved.

**Reply by CagriAldemir** [reply]
In which version was it resolved? I am using the latest version but still this problem is happening.

**Reply by rboadalabelgrup** [reply]
I'm using the latest version of vscode and copilot and this error still remains.

**Reply by isax5** [reply]
I'm using Codeium instead of Copilot and I also have the same bug from this week

**Reply by rushil1510** [reply]
I'm facing this bug too

**Reply by NicolasTournier** [reply]
Me also VSCODE v1.95.3 COPILOT v1.246.0 Win 10

**Comment by efreibe** [comment]
Actually I'm using version v1.77.9225

**Comment by ccrvlh** [comment]
Same here:

macOS 11.7.4
VSCode 1.76.2
Copilot Extension v1.78.9758

**Reply by CagriAldemir** [reply]
Can you share your extension list? Maybe it's because of an extension.

**Comment by svivian** [comment]
This is still not fixed for me. It seems to happen shortly after creating a new file - i.e. pressing Ctrl+N, writing a class, then saving it. Soon after the "Untitled-1" will appear. I've testing with extensions disabled and it still occurs so it's not related to that.

GitHub Copilot v1.89.156
VS Code 1.79.0
MacOS 13.4

Is there somewhere to actually report bugs in Copilot?

**Reply by Mariownyou** [reply]
did you manage to fix this?

**Reply by svivian** [reply]
No, it's still happening regularly all these months later. Seems like nobody at Github actually uses their own software.

**Comment by j4k0xb** [comment]
Same issue: 
- https://github.com/microsoft/vscode/issues/187694
- https://github.com/microsoft/vscode/issues/176187
- https://github.com/microsoft/vscode/issues/202789
- https://github.com/microsoft/vscode/issues/154881
- https://github.com/orgs/community/discussions/73722
- https://github.com/orgs/community/discussions/73781
- https://github.com/orgs/community/discussions/39094

This workaround reliably fixes it (until the next update 😕), modify the copilot extension bundle (linux: `~/.vscode/extensions/github.copilot-<VERSION>/dist/extension.js`, windows: `%USERPROFILE%\.vscode`):
`>=1.148.0`:
```diff
return this.getTextDocumentWithValidation(t).then(n=>{
- if(n.status==="valid")
+ if(n.status==="valid" && n.document.uri.scheme !== "untitled")
```

automatically patch via a command:
```sh
sed -i --regexp-extended 's/if\((\w+)\.status==="valid"\)/if(\1.status==="valid" \&\& \1.document.uri.scheme !== "untitled")/' ~/.vscode/extensions/github.copilot-*/dist/extension.js
```

**Reply by Mariownyou** [reply]
how would you do this on Mac since copilot is bundled

**Reply by j4k0xb** [reply]
should be the same on mac and linux

**Comment by svivian** [comment]
Shouldn’t it be a case of updating the `copilotIgnoreManager.isIgnored` function to check for `e.path === 'Untitled-1'` ? (I’m just guessing, haven’t tested)

**Reply by j4k0xb** [reply]
yes that will work as well, I just chose it based on the simplest place to find (there are multiple `isIgnored` methods)

**Comment by CagriAldemir** [comment]
Did @j4k0xb's suggestion solve your problem or does the problem still exist? I'm asking to be sure if the problem is with the copilot extension.

**Reply by Mariownyou** [reply]
it still exists

**Comment by svipas** [comment]
This problem is 100% because of Copilot extension. Disabling extension solves the issue. Copilot randomly creates Untitled files for me. This issue exists for a very long time.

**Reply by Mariownyou** [reply]
how would you suggest solving it

**Comment by Mariownyou** [comment]
I would suggest setting `"files.defaultLanguage": "plaintext",` but problem still exists if you will change syntax of new file

**Comment by davecheney** [comment]
Duplicate of https://github.com/orgs/community/discussions/73722

**Reply by j4k0xb** [reply]
why was *this* closed as resolved?
the linked discussion is the duplicate (1:1 copy pasted but without the videos)

**Comment by sethvoltz** [comment]
I'm encountering this issue still, the only difference is the `untitled-1` file opens behind the current tab.

**Reply by akashagarwal7** [reply]
https://github.com/orgs/community/discussions/45269#discussioncomment-12848144

**Comment by romanrabodzei** [comment]
I have the same problem.
Copilot extension: v1.187.0
VSCode: v1.89.0

**Comment by YuryLandau** [comment]
I don't have the copilot extension installed, but I'm having this problem.
VSCode: 1.90.1

**Reply by akashagarwal7** [reply]
https://github.com/orgs/community/discussions/45269#discussioncomment-12848144

**Comment by tseijp** [comment]
+1

**Comment by pageagency** [comment]
I too am having this issue, except I get two untitled docs. If i close them, they immediately open back up. 
![image](https://github.com/user-attachments/assets/d3cae6ac-7b9a-45ab-9974-28da0b444dd0)

**Comment by rboadalabelgrup** [comment]
In my case, the problem was solved by using this configuration:

`"workbench.editor.languageDetection": false`

**Reply by ru-Spicker** [reply]
It didn't work in my case.

**Comment by Tamerlanchiques** [comment]
Reporting same issue on latest MacOS Sequoia 15.1.1, VS Code Version: 1.95.3 (Universal)

**Comment by waldnzwrld** [comment]
Same issue still happening.

**Comment by cnydo** [comment]
still has the same problem:
- version: 1.97.0

**Comment by akashagarwal7** [comment]
I have this issue happening since at least a couple of years on MacOS. One of the reasons seems to be saving a file with an extension that's supported by VSCode (.md, .js, .json, etc.). Saving file with no extension or an extension like `.asdasd` doesn't causes this.

**Comment by nandoblanco** [comment]
this just started happening to me and i'm on the latest and greatest :(

**Comment by LuisCarrol** [comment]
LOL 3 years? this just started a couple months ago and it's infuriating. Google'd it and lo and behold, guess who's to blame.

Diagnosing bugs in vscode through the ages:
2020: bisect it
2023: google it
2026: disable copilot

**Comment by Ste1io** [comment]
actually way easier to repro than the way OP described even. Change language mode with text in an untitled editor and close file. Only happens in workspaces.

Spelling it out for the lumnut copilot vibe-devs who are still clueless three years later:

- Open a workspace. It must be a workspace.
- Now open a new untitled text window (`CTRL`+`N`)
- Now paste some text (JSON works) to trigger a language mode change (or set the language mode yourself)
- Now close the untitled window, selecting "No" in the save prompt

Even better, open `C:\Users\vibecoder\AppData\Roaming\Code\Backups\{workspaceIdentifier.id}\untited` and watch the fun happen in windows exploder while you play wack-a-tab.

(Direct link to that directory is in `C:\Users\vibecoder\AppData\Roaming\Code\User\globalStorage\storage.json` listed for each open vscode instance).
